Document significantly more SDK stuff (#483)
discord-userdoccers/discord-userdoccers: pages/resources/store.mdx

| Field               | Type   | Description                                                           |
| ------------------- | ------ | --------------------------------------------------------------------- |
| checkout_session_id | string | A client-generated UUID used to identify the current checkout session |

> <RouteHeader method="GET" url="/partner-sdk/guilds/{guild.id}/application-storefront">

Reviewer: i think we need a new resource for ts 💔

| ------------------- | ------ | --------------------------------------------------------------------- |
| checkout_session_id | string | A client-generated UUID used to identify the current checkout session |

<RouteHeader method="GET" url="/applications/storefront/interactions/premium-button/{store_listing.id}">
  Get Application Storefront for Premium Button
</RouteHeader>

Returns information about store listing, relevant SKUs and subscription plans.

###### Response Body

| Field               | Type                                                         | Description            |
| ------------------- | ------------------------------------------------------------ | ---------------------- |
| skus?               | array[[SKU](#sku-object) object]                             | The relevant SKUs      |
| store_listings      | array[[store listing](#store-listing-object) object]         | The store listings     |
| subscription_plans? | array[[subscription plan](#subscription-plan-object) object] | The subscription plans |

<RouteHeader method="GET" url="/partner-sdk/guilds/{guild.id}/application-storefront">
  Get Guild Application Storefront
</RouteHeader>

> Returns a [storefront](#storefront-object) object for the application of the given guild ID.

Reviewer: ```suggestion
Returns a [storefront](#storefront-object) object for the application linked to the given guild ID.
```

| skus_to_user_ids  | map[snowflake, [application SKU recommendation](#application-sku-recommendation-structure) object] | A mapping of SKU IDs to recommended users |
| application       | partial [application](/resources/application#application-object) object                            | The application                           |

###### Application SKU Recommendation Structure

| Field   | Type      | Description                                                                                  |
| ------- | --------- | -------------------------------------------------------------------------------------------- |
| user_id | snowflake | The ID of the user                                                                           |
| reason  | string    | The [reason why the SKU was recommended to the user](#application-sku-recommendation-reason) |

###### Application SKU Recommendation Reason

| Value          | Description                                |
| -------------- | ------------------------------------------ |
| WISHLIST       | SKU is recommended as a wishlist item      |

> | RECOMMENDATION | SKU is recommended as generic in-game item |

Reviewer: ```suggestion
| WISHLIST       | User has the SKU wishlisted |
| RECOMMENDATION | SKU is recommended for the user |
```

###### JSON Params

| Field                      | Type                                                                    | Description                                      |
| -------------------------- | ----------------------------------------------------------------------- | ------------------------------------------------ |
| title                      | string                                                                  | The title of the storefront (max 256 characters) |
| logo_asset_id?             | snowflake                                                               | The ID of the logo asset                         |
| light_theme_logo_asset_id? | snowflake                                                               | The ID of the logo asset for light theme         |
| pages                      | array[[storefront page](#storefront-page-structure) object]             | The pages of the storefront (max 5)              |

<RouteHeader method="DELETE" url="/partner-sdk/applications/{application.id}/storefront">
  Remove Application Storefront
</RouteHeader>

> Removes storefront from the given application ID. Returns a 204 empty response on success.

Reviewer: ```suggestion
Deletes the application's storefront. User must be the owner of the application or developer of the owning team.
```

<RouteHeader method="GET" url="/partner-sdk/guilds/{guild.id}/application-storefront/skus/{sku.id}">
  Get Guild Application SKU Storefront
</RouteHeader>

> Returns storefront information about application SKU.

Reviewer: ```suggestion
Returns the storefront associated with the given SKU ID.
```

###### Response Body

| Field         | Type                                             | Description       |
| ------------- | ------------------------------------------------ | ----------------- |
| store_listing | [store listing](#store-listing-object) object    | The store listing |
| assets        | array[[store asset](#store-asset-object) object] | The assets        |

<RouteHeader method="GET" url="/partner-sdk/guilds/{guild.id}/application-storefront/announcement">
  Get Guild Application Storefront Announcement
</RouteHeader>

> Returns information about application storefront's announcement.

Reviewer: ```suggestion
Returns the most recent announcement on the storefront for the application linked to the given guild ID.
```
